src/gui/tray/Window.qml

```diff
@@ -403,67 +403,59 @@ Window {
                     }
                 }
 
-                // Filler between account dropdown and header app buttons
-                Item {
-                    id: trayWindowHeaderSpacer
-                    Layout.fillWidth: true
-                }
-
-                HeaderButton {
-                    id: openLocalFolderButton
+                RowLayout {
+                    id: openLocalFolderRowLayout
+                    height: Style.trayWindowHeaderHeight
+                    width: height
+                    anchors.left: accountControlRowLayout.right
+                    anchors.top: accountControlRowLayout.top
+                    spacing: 0
+                    Layout.preferredWidth:  Style.trayWindowHeaderHeight
+                    Layout.preferredHeight: Style.trayWindowHeaderHeight
+                    
+                    HeaderButton {
+                        id: openLocalFolderButton
+                        visible: UserModel.currentUser.hasLocalFolder
+                        icon.source: "qrc:///client/theme/white/folder.svg"
+                        onClicked: UserModel.openCurrentAccountLocalFolder()
+                    }
+                                             
+                    Rectangle {
+                        id: folderStateIndicatorBackground
+                        width: Style.folderStateIndicatorSize
+                        height: width
+                        anchors.top: openLocalFolderButton.verticalCenter
+                        anchors.left: openLocalFolderButton.horizontalCenter
+                        color: Style.ncBlue
+                        radius: width*0.5
+                    }
 
-                    visible: UserModel.currentUser.hasLocalFolder
-                    icon.source: "qrc:///client/theme/white/folder.svg"
-                    onClicked: UserModel.openCurrentAccountLocalFolder()
+                    Image {
+                        id: folderStateIndicator
+                        source: UserModel.isUserConnected(UserModel.currentUserId)
+                                ? Style.stateOnlineImageSource
+                                : Style.stateOfflineImageSource
+                        cache: false
+                        x: folderStateIndicatorBackground.x

```

Comment:
Explicit setting of x, y, width and height is discouraged in the Layout documentation to avoid running into binding loops.
You may also put it as a child of the HeaderButton.